File: 2023-2024/sourcecode/mod1/lab4/32TabLayoutViewPager2NavigationBasicsJava/app/src/main/java/com/example/a32_tablayoutviewpager2navigationbasics_java/TemplateFragment.java
package com.example.a32_tablayoutviewpager2navigationbasics_java;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.a32_tablayoutviewpager2navigationbasics_java.databinding.FragmentTemplateBinding;

public class TemplateFragment extends Fragment {

    private FragmentTemplateBinding binding;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentTemplateBinding.inflate(inflater);

        Bundle args = getArguments();
        if (args != null) {
            binding.textView.setText("Fragment " + args.getInt(PagerAdapter.arg));
        }

        return binding.getRoot();
    }
}

File: 2023-2024/sourcecode/mod1/lab4/31TabLayoutViewPager2NavigationBasicsKotlin/app/src/main/java/com/example/a31_tablayoutviewpager2navigationbasics_kotlin/TemplateFragment.kt
package com.example.a31_tablayoutviewpager2navigationbasics_kotlin

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.example.a31_tablayoutviewpager2navigationbasics_kotlin.databinding.FragmentTemplateBinding

class TemplateFragment : Fragment() {

    private lateinit var binding: FragmentTemplateBinding

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentTemplateBinding.inflate(inflater)

        arguments?.takeIf { it.containsKey(arg) }?.apply {
            binding.textView.text = "Fragment ${getInt(arg)}"
        }

        return binding.root
    }
}